content/docs/philosophy/goals.md

```diff
@@ -0,0 +1,72 @@
+---
+title: Goals and Non-Goals
+sort_rank: 3
+---
+
+# Goals
+
+## Resilience
+
+First and foremost, Prometheus must be resilient in operation.
+
+
+## Reliable alerting
+
+As a monitoring system, Prometheus is being relied upon to alert humans that
+they need to take action in order to prevent undesired system state.
+
+Thus, its most important function is to keep the pipeline of ingestion, rule
+evaluation, and alert hand-off working.
+
+The second most important function is to give humans context about these alerts
+by allowing access to the most recent data Prometheus ingested
+
+### Resulting design decisions and patterns
+
+Note that this goal might result in widely different design decisions and thus
+operational patterns for different parts of our ecosystem:
+
+For Prometheus itself, this means running every instance as an island of data
+completely detached from every other instance, except for optional federation.
+
+For Alertmanager on the other hand, it means the exact opposite: meshing all
+instances closely together, sharing knowledge about alerts and their
+notifications.
+
+## Simple operation
+
+Operation of Prometheus should be as easy and failure-tolerant as possible. We
+try to put required complexity into earlier phases, going through them less
+often and ideally still while under the control of a smaller subset of people.
+
+One example of this would be the preference of statically linked binaries over
+dynamically-built ones.
+
+## Automation
+
+Computers are good at doing the same thing over and over again, and quickly.
+Humans tend to be better at creative tasks.
+
+Prometheus will always strive to automate away all tasks whenever possible
```

Comment:
I don't see how this ties into our philosophy at all. Any automation like this is outside the scope of Prometheus.

What we do have is a belief that alerts going to humans should require intelligent human action.

Reply:
I agree. Maybe RichiH was trying to say that SD (because it's mentioned as an example) dynamically discovering targets is more "automatic" than e.g. an old-style static Nagios hosts configuration? But the section also feels a bit unclear / out of place to me.

Reply:
+1. The automation piece fits somewhere as advice - don't do this without CM, etc - maybe this is more about configuration and dynamism?

Reply:
Seems more like a place to like to My Philosophy on Alerting.

I'd also caution on putting automation as a key goal for users, elimination is better.